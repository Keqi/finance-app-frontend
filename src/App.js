import { Routes, Route } from 'react-router-dom';

import Data from './components/Data'
import Charts from './components/Charts'
import InsertRecord from './components/InsertRecord'

function App() {
  return (
    <>
      <Routes>
         <Route path="/" element={<Data />} />
         <Route path="/charts" element={<Charts />} />
         <Route path="/insert-record" element={<InsertRecord />} />
      </Routes>
   </>
  );
}

export default App;
