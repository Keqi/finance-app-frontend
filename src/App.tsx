import { Routes, Route } from 'react-router-dom';

import Navbar from './components/navbar/Navbar';
import Data from './components/Data'
import Charts from './components/Charts'
import InsertRecord from './components/InsertRecord'

function App() {
  return (
    <>
      <Navbar/>

      <div className="p-5">
        <Routes>
           <Route path="/" element={<Data />} />
           <Route path="/charts" element={<Charts />} />
           <Route path="/insert-record" element={<InsertRecord />} />
        </Routes>
      </div>
   </>
  );
}

export default App;
