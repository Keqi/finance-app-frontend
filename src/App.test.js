import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import { BrowserRouter } from 'react-router-dom';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <App/>
    </BrowserRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});