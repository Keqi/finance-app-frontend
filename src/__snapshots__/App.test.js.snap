// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand navbar-light bg-light"
  >
    <div
      className="container-fluid"
    >
      <span
        className="navbar-brand mr-5 h1"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-dollar-sign fa-xl "
          data-icon="dollar-sign"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M160 0c17.7 0 32 14.3 32 32V67.7c1.6 .2 3.1 .4 4.7 .7c.4 .1 .7 .1 1.1 .2l48 8.8c17.4 3.2 28.9 19.9 25.7 37.2s-19.9 28.9-37.2 25.7l-47.5-8.7c-31.3-4.6-58.9-1.5-78.3 6.2s-27.2 18.3-29 28.1c-2 10.7-.5 16.7 1.2 20.4c1.8 3.9 5.5 8.3 12.8 13.2c16.3 10.7 41.3 17.7 73.7 26.3l2.9 .8c28.6 7.6 63.6 16.8 89.6 33.8c14.2 9.3 27.6 21.9 35.9 39.5c8.5 17.9 10.3 37.9 6.4 59.2c-6.9 38-33.1 63.4-65.6 76.7c-13.7 5.6-28.6 9.2-44.4 11V480c0 17.7-14.3 32-32 32s-32-14.3-32-32V445.1c-.4-.1-.9-.1-1.3-.2l-.2 0 0 0c-24.4-3.8-64.5-14.3-91.5-26.3c-16.1-7.2-23.4-26.1-16.2-42.2s26.1-23.4 42.2-16.2c20.9 9.3 55.3 18.5 75.2 21.6c31.9 4.7 58.2 2 76-5.3c16.9-6.9 24.6-16.9 26.8-28.9c1.9-10.6 .4-16.7-1.3-20.4c-1.9-4-5.6-8.4-13-13.3c-16.4-10.7-41.5-17.7-74-26.3l-2.8-.7 0 0C119.4 279.3 84.4 270 58.4 253c-14.2-9.3-27.5-22-35.8-39.6c-8.4-17.9-10.1-37.9-6.1-59.2C23.7 116 52.3 91.2 84.8 78.3c13.3-5.3 27.9-8.9 43.2-11V32c0-17.7 14.3-32 32-32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
      <div
        className="collapse navbar-collapse d-flex justify-content-start"
      >
        <ul
          className="navbar-nav mr-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/"
              onClick={[Function]}
            >
              Data
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/charts"
              onClick={[Function]}
            >
              Charts
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/insert-record"
              onClick={[Function]}
            >
              Insert record
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="p-5"
  >
    <div
      className="Data"
    >
      <table
        className="table table-bordered mt-4"
      >
        <thead>
          <tr
            className="text-center"
          >
            <th
              className="fs-4"
              scope="col"
            />
            <th
              className="fs-4"
              colSpan={4}
              scope="col"
            >
              ETF (USD)
            </th>
            <th
              className="fs-4"
              colSpan={4}
              scope="col"
            >
              Bonds (PLN)
            </th>
            <th
              className="fs-4"
              colSpan={3}
              scope="col"
            >
              Distribution
            </th>
            <th
              className="fs-4"
              colSpan={5}
              scope="col"
            >
              Total PLN
            </th>
            <th
              className="fs-4"
              colSpan={1}
              scope="col"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            className="text-center"
          >
            <td
              className="fw-bolder"
            >
              Date
            </td>
            <td
              className="fw-bolder"
            >
              Capital
            </td>
            <td
              className="fw-bolder"
            >
              Total
            </td>
            <td
              className="fw-bolder"
            >
              Profit (USD)
            </td>
            <td
              className="fw-bolder"
            >
              Profit %
            </td>
            <td
              className="fw-bolder"
            >
              Capital
            </td>
            <td
              className="fw-bolder"
            >
              Total
            </td>
            <td
              className="fw-bolder"
            >
              Profit (USD)
            </td>
            <td
              className="fw-bolder"
            >
              Profit %
            </td>
            <td
              className="fw-bolder"
            >
              ETF %
            </td>
            <td
              className="fw-bolder"
            >
              Bonds %
            </td>
            <td
              className="fw-bolder"
            >
              USD/PLN
            </td>
            <td
              className="fw-bolder"
            >
              Capital
            </td>
            <td
              className="fw-bolder"
            >
              Total
            </td>
            <td
              className="fw-bolder"
            >
              Profit PLN
            </td>
            <td
              className="fw-bolder"
            >
              Profit %
            </td>
            <td
              className="fw-bolder"
            >
              Accumulated Inflation
            </td>
            <td
              className="fw-bolder"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
]
`;
